pkg/build/api/types.go

```diff
@@ -178,6 +188,26 @@ type GitHubWebHookCause struct {
 	Secret string
 }
 
+// GitLabWebHookCause has information about a GitLab webhook that triggered a
+// build.
+type GitLabWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
+
+// BitbucketWebHookCause has information about a Bitbucket webhook that triggered a
+// build.
+type BitbucketWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
```

Comment:
hm.  @smarterclayton slapped our hand for an inadvertent protobuf change somewhere else so that's why I was averse to it here.  Hopefully he can weigh in..i'd certainly be happy if we can make all 4 of these structs consistent.